src/lsdb/dask/merge_map_catalog_data.py

```diff
@@ -0,0 +1,124 @@
+# pylint: disable=duplicate-code
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Callable, Tuple
+
+import dask
+import nested_dask as nd
+import nested_pandas as npd
+from hats.catalog import TableProperties
+from hats.pixel_math import HealpixPixel
+from hats.pixel_tree import PixelAlignment, PixelAlignmentType
+from hats.pixel_tree.pixel_alignment import align_with_mocs
+
+from lsdb.dask.merge_catalog_functions import (
+    align_and_apply,
+    construct_catalog_args,
+    filter_by_spatial_index_to_pixel,
+    get_healpix_pixels_from_alignment,
+)
+from lsdb.types import DaskDFPixelMap
+
+if TYPE_CHECKING:
+    from lsdb.catalog import Catalog, MapCatalog
+
+
+# pylint: disable=too-many-arguments, unused-argument
+@dask.delayed
+def perform_merge_map(
```

Comment:
Given that this performs a `merge_asof`, why not call it `perform_merge_maps_asof` or similar?  Is `asof` pretty obviously implied through this family of functions?  (Seems like it may be.)

Reply:
This doesn't perform a `merge_asof`, but I had copy/pasted the `merge_asof` as a starting point, and never updated the docstrings!

This functionality is more akin to a custom cross-match, in that some custom function will be applied to spatially matching pairs of partitions.